tests/Extension/Interfaces/InterfaceMethodSignatureTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Zephir.
+ *
+ * (c) Phalcon Team <team@zephir-lang.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Extension\Interfaces;
+
+use PHPUnit\Framework\TestCase;
+use Stub\Interfaces\ImplementInt;
+use Stub\Interfaces\ImplementInterface;
+
+final class InterfaceMethodSignatureTest extends TestCase
+{
+    public function testImplementInterfaceInMethodSignature(): void
+    {
+        $class = new ImplementInt();
+
+        $this->assertNull($class->get());
+        $class->set(1);
+        $this->assertSame(1, $class->get());
+
+        $error = '';
+        try {
```

Comment:
Better to use native PHPUnit methods for Exceptions instead of try/catch and string matches.

For example:
```php
$this->expectException(\TypeError::class);
$this->expectExceptionMessageMatches('/^must be of type int, bool given/');

$class->set(true);
```